File: frontend/src/api/token.ts
import { apiGet, apiPost, apiDelete } from './http'

export type TokenAuthType = 'xtoken' | 'basic'

export interface TokenItem {
  token: string
  issuedAt: string
  expiresAt: string
  revoked: boolean
  issuedByApi: string | null
  meta: Record<string, unknown>
}

export interface TokenCreate {
  token: string
  expiresAt: string
  authType?: TokenAuthType
  issuedByApi?: string | null
  meta?: Record<string, unknown>
}

export interface TokenListParams {
  revoked?: boolean | null
  notExpired?: boolean
  page?: number
  pageSize?: number
}

export interface TokenListResult {
  items: TokenItem[]
  total: number
  page: number
  pageSize: number
}

export const tokenApi = {
  list: (params?: TokenListParams): Promise<TokenListResult> =>
    apiGet('/tokens', params as Record<string, unknown> | undefined),

  create: (data: TokenCreate): Promise<TokenItem> => apiPost('/tokens', data),

  revoke: (token: string): Promise<TokenItem> =>
    apiPost(`/tokens/${encodeURIComponent(token)}/revoke`, {}),

  deleteExpired: (): Promise<{ deleted: number }> => apiDelete('/tokens/expired'),
}

File: backend/app/admin/schemas/token.py
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, Field

from ._base import CamelModel


class TokenCreate(CamelModel):
    token: str = Field(..., min_length=1, max_length=500)
    expires_at: datetime
    auth_type: str = Field(default="xtoken", max_length=20)
    issued_by_api: Optional[str] = Field(default=None, max_length=500)
    meta: dict[str, Any] = Field(default_factory=dict)


class TokenRevoke(CamelModel):
    ...


class TokenItem(CamelModel):
    token: str
    issued_at: datetime
    expires_at: datetime
    revoked: bool
    issued_by_api: Optional[str]
    meta: dict[str, Any] = Field(default_factory=dict)


class TokenListData(CamelModel):
    items: list[TokenItem]
    total: int
    page: int
    pageSize: int


class TokenListResponse(CamelModel):
    code: int = 0
    data: TokenListData
    message: str = "ok"


class TokenItemResponse(CamelModel):
    code: int = 0
    data: TokenItem
    message: str = "ok"
